miio/chuangmi_camera.py

```diff
@@ -375,13 +379,29 @@ def set_nas_config(
         video_retention_time: NASVideoRetentionTime = NASVideoRetentionTime.Week,
     ):
         """Set NAS configuration."""
-        if share is None:
-            share = {}
-        return self.send(
-            "nas_set_config",
-            {
-                "state": state,
-                "sync_interval": sync_interval,
-                "video_retention_time": video_retention_time,
-            },
-        )
+        share_obj = None
+        share = urlparse(share)
+        if share.scheme == "smb":
+            ip = socket.gethostbyname(share.hostname)
+            reversed_ip = ".".join(reversed(ip.split(".")))
+            addr = int(ipaddress.ip_address(reversed_ip))
```

Comment:
Is reversing the address really necessary?

Reply:
Yes, I use MiHome App set the smb config, then get config by miiocli, and I see the address is stored reversed.